cirq/ops/eigen_gate.py

```diff
@@ -134,8 +134,8 @@ def matrix(self) -> np.ndarray:
         if self.is_parameterized():
             raise ValueError("Parameterized. Don't have a known matrix.")
         e = cast(float, self._exponent)
-        return np.sum(1j**(half_turns * e * 2) * component
-                      for half_turns, component in self._eigen_components())
+        return sum(1j**(half_turns * e * 2) * component
```

Comment:
Done.

Though I'm not sure what numpy was actually doing behind the scenes. It might have just been implicitly wrapping the generator in a list, or it might have been summing using the generator. It's a tradeoff between space and speed, possibly.